src/core/json/json_object.cc

```diff
@@ -62,4 +138,45 @@ JsonType DeepCopyJSON(const JsonType* j) {
   return std::move(deserialized.value());
 }
 
+void Defragment(JsonType& j, PageUsage* page_usage) {
```

Comment:
These new functions can also live somewhere else, eg compact object or similar, then we would not have link `dfly_page_usage` with json library. I found it more logical to place it here with other json code.